util/threadpool_imp.cc

```diff
@@ -189,10 +229,14 @@ void ThreadPoolImpl::Impl::BGThread(size_t thread_id) {
     // Wait until there is an item that is ready to run
     std::unique_lock<std::mutex> lock(mu_);
     // Stop waiting if the thread needs to do work or needs to terminate.
+    // increase num_waiting_ once this task has finished
```

Comment:
Should this be "increase num_waiting_ once the thread starts waiting" to be consistent with "decrease num_waiting_ once the thread is not waiting" below?

nit: Start with a capitalized letter in the comments to be consistent with others such as  "Increase ..." or "Decrease ..." (below)

Reply:
That was my typo. Thanks for pointing out this.